### PR TITLE
disttask: dispatcher should update global task and add new tasks in a txn

### LINKED ISSUE #42848: disttask: dispatcher should update global task and add new tasks in a txn

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #42848

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by wjhuang2016 — APPROVED** [review]

File: disttask/framework/storage/task_table.go [hunk 1]
```diff
@@ -169,19 +197,7 @@ func (stm *TaskManager) GetNewGlobalTask() (task *proto.Task, err error) {
 
 // UpdateGlobalTask updates the global task.
 func (stm *TaskManager) UpdateGlobalTask(task *proto.Task) error {
```
Comment (zimulala) [inline disttask/framework/storage/task_table.go:199]: Could we remove this function?

File: disttask/framework/storage/task_table.go [hunk 2]
```diff
@@ -339,3 +355,35 @@ func (stm *TaskManager) GetSchedulerIDsByTaskID(taskID int64) ([]string, error)
...
+			}
+		})
+
+		subtaskState := proto.TaskStatePending
+		if isSubtaskRevert {
+			subtaskState = proto.TaskStateRevertPending
+		}
+
+		for _, subtask := range subtasks {
+			_, err = execSQL(stm.ctx, se, "insert into mysql.tidb_background_subtask(task_key, exec_id, meta, state, type, checkpoint) values (%?, %?, %?, %?, %?, %?)",
```
Comment (zimulala) [inline disttask/framework/storage/task_table.go:380]: Could we consider batch insert or add a  TODO?

Comment (lcwangchao) [reply disttask/framework/storage/task_table.go:376]: added some comments

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**